docs/design/references/METADATA.md

```diff
@@ -0,0 +1,22 @@
+# `nupm`'s package metadata reference
+
+## Required attributes
+- `author`: Name of the developer/organization/etc.
+- `files`: List of records of files being part of the package. Records reference:
+  - `checksum`: SHA256 sum of the file
+  - `name`: File path (relative to the package scope)
+  - `raw-url`: `GET`table link to the file
+  - `supported-os`: Exactly like in the top-level metadata
+- `name`: Package name
+- `nu-version`: Supported Nushell version. Just like `dependencies` might be either exact version or some matcher like "greater than 1.70"
+- `short-description`: Short info about the package, displayed by default
+- `supported-os`: Operating systems supported by the package, the most broad possibility: `{"arch": ["*"], "family": ["*"], "name": ["*"]}`. Matched by `$nu.os-info`
+- `url`: Package website/GitHub repository. Basically a place where one can find some additional info about the package
+- `version`: Version of the package. Semantic versioning is advised to enable users to have more generic requirements
+
+## Optional attribtues
+- `dependencies`: Packages needed by the package — versions have to be specified. e.g. `[nupm/0.7.0]`. Semantic versioning is also supported: `[nupm/~0.7]`
+- `installer`: Name of a script (relative to the package scope) that will install the package instead (or in addition to) of default `nupm` logic
+- `keywords`: List of keywords used by `nupm search` in addition to `name`
+- `pre-install-hook`: Name of a script (relative to the package scope) to be executed before installation. Might be simple pre-install message or more complex logic
+- `post-install-hook`: Name of a script (relative to the package scope) to be executed after installation. Might be simple post-install message or more complex logic
```

Comment:
```suggestion
```

We can just get rid of it, as mentioned on Discord and in one thread

Reply:
done in ad6b0cd8a06722ce05f90a2f67bf782f40c87010